util/log/file.go

```diff
@@ -127,3 +134,51 @@ func create(tag string, t time.Time) (f *os.File, filename string, err error) {
 	}
 	return nil, "", fmt.Errorf("log: cannot create log: %v", lastErr)
 }
+
+// A LogFileInfo holds the filename and size of a log file.
+type LogFileInfo struct {
+	Name         string // base name
+	SizeBytes    int64
+	ModTimeNanos int64 // most recent mode time in unix nanos
+}
+
+// ListLogFiles returns a slice of LogFileInfo structs for each log
+// file on the local node, in any of the configured log directories.
+func ListLogFiles() ([]LogFileInfo, error) {
+	var results []LogFileInfo
+	for _, dir := range logDirs {
+		infos, err := ioutil.ReadDir(dir)
+		if err != nil {
+			return results, err
+		}
+		for _, info := range infos {
+			// Only list regular files & ensure that files we locate here match the log file regexp.
+			if info.Mode()&os.ModeType == 0 && logFileRE.MatchString(info.Name()) {
+				results = append(results, LogFileInfo{
+					Name:         info.Name(),
+					SizeBytes:    info.Size(),
+					ModTimeNanos: info.ModTime().UnixNano(),
+				})
+			}
+		}
+	}
+	return results, nil
+}
+
+// GetLogReader returns a reader for the specified filename, taking
+// care to make the filename an absolute path according to the log
+// directory, if necessary.
+func GetLogReader(filename string) (io.ReadCloser, error) {
+	if path.IsAbs(filename) {
+		return os.Open(filename)
```

Comment:
GetLogReader should not be able to open non-log files; it should be an error to pass an absolute path (or even a path containing any slashes) here.